swagger.yaml

```diff
@@ -7013,9 +7041,46 @@ definitions:
         description: Indication that the version is the recommended one.
       support_level:
         type: string
-        enum: [beta, production, maintenance]
+        enum: [beta, production, maintenance, end of life]
         description: Level of support of the version.
 
+  upgrade-channel:
+    type: object
+    required:
+      - cpu_architecture
+      - channels
+    properties:
+      cpu_architecture:
+        type: string
+        enum: ['x86_64', 'aarch64', 'arm64','ppc64le','s390x','multi']
+        description: The CPU architecture of the image.
+        x-go-custom-tag: gorm:"default:'x86_64'"
+      channels:
+        type: array
+        items:
+          type: string
+          enum: [stable, candidate]
```

Comment:
let's add `fast` and `eus` (in general, all possible channels: https://docs.openshift.com/container-platform/4.14/updating/understanding_updates/understanding-update-channels-release.html )

Reply:
I was contemplating whether to use `eus` and `fast` as well. I decided to not incluse them as IMHO they don't give us more functionality, but they do add complexity to  `releasesources` logic, so I decided to omit them. all releases in `fast` channel will soon be in `sable` as well (probably, or they shouldn't), and `eus` doesn't really say anything extra for us because every release in `eus` is also in `stable` .
WDYT?

Reply:
What kind of complexity does it add?

Reply:
Thinking about this again I think after the recent changes we don't need to store `channel` field in the DB, so it reduces the complexity of adding these channels.